File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

public static class DifferenceOfSquares
{
    static int Square(this int value) => value * value;
    static IEnumerable<int> GetRange(this int max) => Enumerable.Range(start: 1, count: max);

    public static int CalculateSquareOfSum(int max) => max.GetRange().Sum().Square()
;
    public static int CalculateSumOfSquares(int max) => max.GetRange().Select(Square).Sum();

    public static int CalculateDifferenceOfSquares(int max) => CalculateSquareOfSum(max) - CalculateSumOfSquares(max);
}

File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  // Put your code here
  int square(int i) => i * i;
  int sum(int a, int b) => a + b;
  List<int> range(int val) => List<int>.generate(val, (int i) => i + 1);
  int sumOfSquares(int val) => range(val).map(square).reduce(sum);
  int squareOfSum(int val) => square(range(val).reduce(sum));
  int differenceOfSquares(int val) => squareOfSum(val) - sumOfSquares(val);
}
